dsgrid/dimension/base_models.py

```diff
@@ -150,6 +149,37 @@ def check_required_dimensions(dimensions, tag):
     check_uniqueness((x.dimension_type for x in dimensions), tag)
 
 
+def check_required_dataset_dimensions(dimensions, tag):
+    """Check that a dataset config contains all required dimensions.
+
+    Parameters
+    ----------
+    dimensions : list
+        list of DimensionReferenceModel
+    tag : str
+        User-defined string to include in exception messages
+
+    Raises
+    ------
+    ValueError
+        Raised if a required dimension is not provided.
+
+    """
+    # dimension_types = {x.dimension_type for x in dimensions}
```

Comment:
We talked about requiring a subset of dimensions, but I'm not sure that any are absolutely required.